tests/test_csc.py

```diff
@@ -0,0 +1,136 @@
+                state = await harness.remote.evt_summaryState.next(flush=False, timeout=STD_TIMEOUT)
+                self.assertEqual(state.summaryState, salobj.State.DISABLED)
+
+                await harness.remote.cmd_enable.start()
+                self.assertEqual(harness.csc.summary_state, salobj.State.ENABLED)
+                state = await harness.remote.evt_summaryState.next(flush=False, timeout=STD_TIMEOUT)
+                self.assertEqual(state.summaryState, salobj.State.ENABLED)
+
+                await harness.remote.cmd_disable.start()
+                self.assertEqual(harness.csc.summary_state, salobj.State.DISABLED)
+                state = await harness.remote.evt_summaryState.next(flush=False, timeout=STD_TIMEOUT)
+                self.assertEqual(state.summaryState, salobj.State.DISABLED)
+
+                await harness.remote.cmd_standby.start()
+                self.assertEqual(harness.csc.summary_state, salobj.State.STANDBY)
+                state = await harness.remote.evt_summaryState.next(flush=False, timeout=STD_TIMEOUT)
+                self.assertEqual(state.summaryState, salobj.State.STANDBY)
+
+        asyncio.get_event_loop().run_until_complete(doit())
+
+    def test_apply_position_limits(self):
+        async def doit():
+            async with Harness(initial_state=salobj.State.STANDBY) as harness:
+                await harness.remote.cmd_start.start(timeout=STD_TIMEOUT)
+                await harness.remote.cmd_enable.start(timeout=STD_TIMEOUT)
+                harness.remote.evt_settingsAppliedPositionLimits.flush()
+                await harness.remote.cmd_applyPositionLimits.set_start(timeout=STD_TIMEOUT, xyMax=12, zMin=-6,
+                                                                       zMax=12, uvMax=8, wMin=-6, wMax=6)
+                event = await harness.remote.evt_settingsAppliedPositionLimits.next(flush=False,
+                                                                                    timeout=STD_TIMEOUT)
+                self.assertEqual(12, event.limitXYMax)
+                self.assertEqual(-6, event.limitZMin)
+                self.assertEqual(12, event.limitZMax)
+                self.assertEqual(8, event.limitUVMax)
+                self.assertEqual(-6, event.limitWMin)
+                self.assertEqual(6, event.limitWMax)
+
+        asyncio.get_event_loop().run_until_complete(doit())
+
+    def test_move_to_position(self):
+        async def doit():
+            async with Harness(initial_state=salobj.State.STANDBY) as harness:
+                await harness.remote.cmd_start.start(timeout=STD_TIMEOUT)
+                await harness.remote.cmd_enable.start(timeout=STD_TIMEOUT)
+                await harness.remote.cmd_moveToPosition.set_start(timeout=STD_TIMEOUT, x=4, y=6, z=3, u=4,
+                                                                  v=3, w=6)
+                await asyncio.sleep(SHORT_TIMEOUT)

```

Comment:
What is this sleep for? If the problem is that you sometimes get timeouts in the next line then I think it would be clearer to increase that timeout, instead.

Reply:
That was a Tiago suggestion.

Reply:
That sleep cannot be doing anything useful here.

A typical use of such a sleep statement is to free up the event loop, but you are already freeing it by awaiting on the next line.

Reply:
Its because the movement command will only complete when a asyncio.wait_for is done. Therefore I need a little extra time for that.

Reply:
In that case I suggest you increase the timeout on the next line by (roughly) the expected move time. It will make it much clearer what is going on. This sleep statement is unnecessary.